Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -576,6 +580,36 @@ private static boolean isDartPatchUri(@NotNull final String uri) {
     return result;
   }
 
+  @Nullable
+  public XDebuggerEvaluator getEvaluator() {
+    XStackFrame frame = getSession().getCurrentStackFrame();
+    if (frame != null) {
+      return frame.getEvaluator();
+    }
+    String isolateId = myLatestCurrentIsolateId;
+    if (isolateId == null) {
+      if (getIsolateInfos().isEmpty()) {
+        return null;
+      } else {
+        // A breakpoint hasn't yet been hit but some isolates have been
+        // started. Use the first one.
+        isolateId = getIsolateInfos().iterator().next().getIsolateId();
+      }
+    }
+    Project project = getSession().getProject();
+    FileEditorManager manager = FileEditorManager.getInstance(project);
+
+    VirtualFile[] files = manager.getSelectedFiles();
```

Comment:
Speaking of the context file selection: I'm not sure there's an easy way to add special UI for this purpose. But it may make sense to move logic that takes the file from the current editor right to the `evaluate()` method. Users may switch editor tabs without closing the Evaluate Expression dialog.

Reply:
Done. I agree that moving the calculation to right at the evaluate call is least surprising.